components/event-bus/internal/knative/util/util.go

```diff
@@ -6,24 +6,27 @@ import (
 )
 
 const (
+	channelNameDelimiter    = "--"
 	defaultChannelNamespace = "kyma-system"
 )
 
 var (
-	replacer = strings.NewReplacer("-", "--", ".", "-dot-")
+	replacer = strings.NewReplacer("-", "-d", ".", "-p")
 )
 
-// The function applies the following rules in order:
-//  * In case there was a '-' or more in any of the argument values, each occurrence of the '-' will be escaped by '--'.
-//  * In case there was a '.' or more in any of the argument values, each occurrence of the '.' will be replaced by the '-dot-' character sequence, because of a limitation
-//    in the current knative version, if the channel name has a '.', the corresponding istio-virtualservice will not be created.
+// The escapeHyphensAndPeriods function applies the following rules in order:
+//  * In case there was a '-' or more in any of the argument values, each occurrence of the '-' will be escaped by '-d'.
+//  * In case there was a '.' or more in any of the argument values, each occurrence of the '.' will be replaced by the '-p' character sequence,
+//    because of a limitation in the current knative version, if the channel name has a '.', the corresponding istio-virtualservice will not be created.
 func escapeHyphensAndPeriods(str *string) string {
 	return replacer.Replace(*str)
 }
 
-// GetChannelName function joins the sourceID, eventType and eventTypeVersion respectively with a '-' as a delimiter.
+// GetChannelName function joins the sourceID, eventType and eventTypeVersion respectively with a '--' as a delimiter.
 func GetChannelName(sourceID, eventType, eventTypeVersion *string) string {
-	return fmt.Sprintf("%s-%s-%s", escapeHyphensAndPeriods(sourceID), escapeHyphensAndPeriods(eventType), escapeHyphensAndPeriods(eventTypeVersion))
+	return fmt.Sprintf("%s%s%s%s%s", escapeHyphensAndPeriods(sourceID), channelNameDelimiter,
+		escapeHyphensAndPeriods(eventType), channelNameDelimiter,
+		escapeHyphensAndPeriods(eventTypeVersion))
 }
 
 // GetKnSubscriptionName joins the kySubscriptionName and kySubscriptionNamespace
```

Comment:
Should we change GetKnSubscriptionName() to be conform with GetChannelName() tpp?

Reply:
Done.